index.js

```diff
@@ -1,8 +1,4 @@
-module.exports = function(str) {
-  if (!str) {
-    return null;
-  }
-
-  var d = new Date(String(str));
+module.exports = function(val) {
+  var d = new Date(val);
```

Comment:
This should still be the case. `new Date(null)` and `new Date(undefined)` will result in `NaN`, and cause `parse-date` to return `null`. I added a test to confirm.